library/lua/gui/widgets.lua

```diff
@@ -808,7 +808,7 @@ local function scrollbar_get_max_pos_and_height(scrollbar)
     local frame_body = scrollbar.frame_body
     local scrollbar_body_height = (frame_body and frame_body.height or 3) - 2
 
-    local height = math.max(1, math.floor(
+    local height = math.max(2, math.floor(
```

Comment:
I'm not sure if this has any practical effect on the scrollbar. It's still beholden to the parent frame. What difference does this change actually make visually?

Reply:
it turns this...
![Capture1](https://user-images.githubusercontent.com/5853667/216965875-0c74fc9c-9577-4f2f-8388-e3b7deff69b0.PNG)

into this...
![Capture3](https://user-images.githubusercontent.com/5853667/216965890-b020cee8-8237-436b-9687-a28b768fe2c0.PNG)

and if you use a custom texture, the scrollbar will always look good
![Capture2](https://user-images.githubusercontent.com/5853667/216965974-b9482618-7ead-4f63-ba8f-b650b21dbe69.PNG)

like, if we wanted 1-tall scrollbar graphics to "look good", we could probably define about 6 more sprites for different states... like so:
(from left to right: 2-tall, current 1-tall, "good-looking" 1-tall, possible implementations of different states of it)
(those red-highlit parts are to show that the goodlooking 1tall overflows into the lower tiles)
![Captur4e](https://user-images.githubusercontent.com/5853667/216967727-8da15142-461c-4891-aeda-7a3a5a72bc09.PNG)

...but! maybe that's just too much of a useless detail to be concerned about. Like, limiting scrollbars' height to 2 lets us *not* implement that crap, *and* keep the ui looking clean, consistent.


worthy of note though, i just noticed a bit of a weird behaviour. When the 'bar visually reaches the bottom of the list, you can still drag it downwards 1 more tile and scroll some more.
[2023-02-06 12-54-06-1.webm](https://user-images.githubusercontent.com/5853667/216968171-006fabc6-5aac-4601-bf99-dbd44515a5d8.webm)

Reply:
>I'm not sure if this has any practical effect on the scrollbar. It's still beholden to the parent frame.

come to think of it, i guess that weird behaviour i described is surely related to what you just said.
I... I, like, honestly just tried changing numbers here and there, and _that one_ i changed _seemed_ to do what i wanted... 😂 so i felt like a supa haka and was all like PULLREQUEST THIS SHIT and here we are.

if we want to do all this 2-tall nonsense, we can probably iron out that behaviour as well, if needed.
or even, if you're really interested in the "goodlooking" system, i could make some 'sheets for those i guess 🤔 i guess it would be like having a 2-tall bar anyway, just with a different look and more logic to it. Not sure if it's worth doing though